packages/react-hooks/src/useScreen/useScreen.ts

```diff
@@ -1,23 +1,42 @@
 import { useEffect, useReducer } from "react";
-import { reducer, initialState, init, actionTypes, State } from "./state";
+import { breakpoints } from "@fremtind/jkl-core";
+import { addMediaQueryListener, getInitialMediaQueryMatch } from "../mediaQueryUtils";
+import { ActionType, reducer, ScreenState } from "./state";
 
-export const useScreen = (): State => {
-    const [device, deviceDispatch] = useReducer(reducer, initialState, init);
+const MEDIA_RULES: Record<keyof ScreenState, string> = {
+    isSmallDevice: `(max-width: ${breakpoints.medium - 1}px)`,
+    isMediumDevice: `(min-width: ${breakpoints.medium}px) and (max-width: ${breakpoints.large - 1}px)`,
+    isLargeDevice: `(min-width: ${breakpoints.large}px) and (max-width: ${breakpoints.xl - 1}px)`,
+    isXlDevice: `(min-width: ${breakpoints.xl}px)`,
+    isPortrait: "(orientation: portrait)",
+    isLandscape: "(orientation: landscape)",
+};
+
+export const useScreen = (): ScreenState => {
+    const initialState: ScreenState = Object.fromEntries(
+        Object.entries(MEDIA_RULES).map(([key, rule]) => [key, getInitialMediaQueryMatch(rule)]),
+    ) as unknown as ScreenState;
```

Comment:
Denne er sikkert fin å memoe, siden det er initiell state det er snakk om:

```suggestion
    const initialState: ScreenState = useMemo(
        () =>
            Object.fromEntries(
                Object.entries(MEDIA_RULES).map(([key, rule]) => [key, getInitialMediaQueryMatch(rule)]),
            ),
        [],
    ) as unknown as ScreenState;
```